src/gui/N2Go_Gui.php

```diff
@@ -310,4 +311,21 @@ private function saveFormType($forms, $formUniqueCode)
             }
         }
     }
+
+    private function verifyResponse($response)
+    {
+        switch($response['response']['code']){
+            case 200:
+                return json_decode($response['body']);
+                break;
+            case 400:
+                $this->disconnect();
+                return false;
+                break;
+            case 401:
+            case 403:
+                return $this->refreshTokens();
```

Comment:
this is good but can goes in an endless loop because **$this->refreshTokens()** calls this method again.
Possible solutions to me are to have a class parameter to check if the verification should happens, or  to pass a parameter to stop **$this->refreshTokens()** calling  this method again. Maybe somthing like:
```php
return $this->refreshTokens(false);
```

and in the refreshTokens something like:
```php
private function refreshTokens($verify= true)
{
...
if ($verify && $this->verifyResponse($responseRaw)){
...
}